src/matrixrtc/MembershipManager.ts

```diff
@@ -997,14 +1017,64 @@ export class MembershipManager
     }
 }
 
-function createInsertActionUpdate(type: MembershipActionType, offset?: number): ActionUpdate {
-    return {
-        insert: [{ ts: Date.now() + (offset ?? 0), type }],
-    };
-}
+/**
+ * Implementation of the Membership manager that uses sticky events
+ * rather than state events.
+ */
+export class StickyEventMembershipManager extends MembershipManager {
+    public constructor(
+        joinConfig: (SessionConfig & MembershipConfig) | undefined,
+        room: Pick<Room, "getLiveTimeline" | "roomId" | "getVersion">,
+        private readonly clientWithSticky: Pick<
+            MatrixClient,
+            | "getUserId"
+            | "getDeviceId"
+            | "sendStateEvent"
+            | "_unstable_sendDelayedStateEvent"
+            | "_unstable_updateDelayedEvent"
+            | "_unstable_sendStickyEvent"
+            | "_unstable_sendStickyDelayedEvent"
+        >,
+        getOldestMembership: () => CallMembership | undefined,
+        sessionDescription: SessionDescription,
+        parentLogger?: Logger,
+    ) {
+        super(joinConfig, room, clientWithSticky, getOldestMembership, sessionDescription, parentLogger);
+    }
 
-function createReplaceActionUpdate(type: MembershipActionType, offset?: number): ActionUpdate {
-    return {
-        replace: [{ ts: Date.now() + (offset ?? 0), type }],
-    };
+    protected clientSendDelayedEvent: (myMembership: EmptyObject) => Promise<SendDelayedEventResponse> = (
+        myMembership,
+    ) =>
+        this.clientWithSticky._unstable_sendStickyDelayedEvent(
+            this.room.roomId,
+            STICK_DURATION_MS,
+            { delay: this.delayedLeaveEventDelayMs },
+            null,
+            EventType.GroupCallMemberPrefix,
+            { msc4354_sticky_key: this.stateKey },
+        );
+
```

Comment:
This method is odd since it takes an empty object as its param and then does not use it.
 - Drop the param
 -  use the param like so

```suggestion
    protected clientSendDelayedEvent: (myMembership: EmptyObject) => Promise<SendDelayedEventResponse> = (
        myMembership,
    ) =>
        this.clientWithSticky._unstable_sendStickyDelayedEvent(
            this.room.roomId,
            STICK_DURATION_MS,
            { delay: this.delayedLeaveEventDelayMs },
            null,
            EventType.GroupCallMemberPrefix,
            {...myMembership, msc4354_sticky_key: this.stateKey },
        );

```

Reply:
Lets just drop the param and call this `clientSendDelayedDisconnectMembership`, which makes it more clear what it does.